### PR TITLE
feat: adds concurrent_insurance_buy go script

### PR DESCRIPTION
**Pull request by Justintime50**
This tool allows a user to purchase insurance (EasyPost shipment or standalone insurance) in bulk by supplying a CSV file with tracking codes. We take in the CSV, prep and validate the data, spin up 20 goroutines concurrently to make requests fast, and await the entire CSV to finish processing before writing an output CSV to disk that contains the status of each request, the time it took, and any error messages. An example of the output can be seen here:

```text
$ EASYPOST_API_KEY=123 CSV=sample.csv ./bulkins
Sending request for EZ7000000007...
Sending request for EZ2000000002...
Sending request for EZ3000000003...
Sending request for EZ1000000001...
Sending request for EZ5000000005...
Sending request for EZ4000000004...
Sending request for EZ6000000006...

Total time elapsed: 663.889417ms
```

Screenshot 2024-08-19 at 4 22 36 PM

### CODE REVIEW

File: community/golang/concurrent_insurance_buy/concurrent_insurance_buy.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+		// Skip header
+		if lineNumber == 0 {
+			lineNumber++
+			continue
+		}
+
+		// Ensure `Amount` column is a valid float
+		_, err = strconv.ParseFloat(record[3], 64)
+		if err != nil {
+			handleGoErr(err)
```
Comment (nwithan8) [inline community/golang/concurrent_insurance_buy/concurrent_insurance_buy.go:112]: nit: `handleGoErr` does a `err != nil` check in itself, don't need to put this behind a nullability check.

**Review by nwithan8 — APPROVED** [review]

File: community/golang/concurrent_insurance_buy/sample.csv [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Reference,Tracking Code,Carrier String,Amount,To Address ID (optional),From Address ID (optional)
```
Comment (jchen293) [inline community/golang/concurrent_insurance_buy/sample.csv:1]:
```suggestion
Reference (optional),Tracking Code,Carrier String,Amount,To Address ID (optional),From Address ID (optional)

```

`reference` is optional when creating standalone insurance

**Review by jchen293 — APPROVED** [review]

File: community/golang/concurrent_insurance_buy/concurrent_insurance_buy.go (hunk 1 above)
Comment (Justintime50) [reply community/golang/concurrent_insurance_buy/concurrent_insurance_buy.go:112]: Nice catch, fixed!

File: community/golang/concurrent_insurance_buy/sample.csv (hunk 2 above)
Comment (Justintime50) [reply community/golang/concurrent_insurance_buy/sample.csv:1]: Indeed, fixed

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**